### PR TITLE
feat(docs): add `RELEASE_PROCESS.md`

### PR DESCRIPTION
**Pull request by Evalir**
Adds a `RELEASE_PROCESS.md` file that describes how the new release process is intended to look.

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+## Step 1 - Testing and benchmarking
+
+Prior to creating a release commit and tagging it for release, we carefully test and benchmark the chosen commit. This is made to ensure that regressions are not made, and measure performance differences across versions.
+
+## Step 2 - Create release commit
+
+Creating a release commit involves the following steps:
+- The `CHANGELOG.md` file is double-checked to be updated, and a new section indicating the new stable version is created, with the changes included.
+- This `CHANGELO.md` is committed along with the tag name, e.g `v1.0.0`.
```
Comment (gakonst) [inline RELEASE_PROCESS.md:15]:
```suggestion
- This `CHANGELOG.md` is committed along with the tag name, e.g `v1.0.0`.
```

Comment (gakonst) [reply RELEASE_PROCESS.md:15]: IMO let's just use the Release Notes on Github for the changelogs - like in Reth. Unless you think we rly need a separate file.

Comment (Evalir) [reply RELEASE_PROCESS.md:15]: Ah fair i'm frankly OK with whichever option we go for. I'm happy to not have to maintain a file manually :)

Comment (Evalir) [reply RELEASE_PROCESS.md]: changed to point out that crate versions need to be bumped instead @gakonst see https://github.com/foundry-rs/foundry/pull/5269/commits/3e4ae681fc908de98fbe54ddb0e20906ef4bc604

**Review by mattsse — APPROVED** [review]

**Pull request merged**